DfciPkg/Application/DfciMenu/DfciCheck429.c

```diff
@@ -0,0 +1,72 @@
+/** @file
+DfciSARecovery.c
+
+Device Firmware Configuration Interface - Stand Alone driver that can be loaded
+at the UEFI Shell
+
+Copyright (C) Microsoft Corporation. All rights reserved.
+SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <Uefi.h>
+
+#include <Library/BaseLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/DebugLib.h>
+#include <Library/HttpLib.h>
+#include <Library/MemoryAllocationLib.h>
+#include <Library/PrintLib.h>
+#include <Library/UefiApplicationEntryPoint.h>
+#include <Library/UefiLib.h>
+
+#include "DfciPrivate.h"
+#include "DfciRequest.h"
+
+// *---------------------------------------------------------------------------------------*
+// * Application Global Variables                                                          *
+// *---------------------------------------------------------------------------------------*
+DFCI_NETWORK_REQUEST  mDfciNetworkRequest = { 0 };
+STATIC CHAR8          mCheck429Url[]      = "http://mikeytbds3.eastus.cloudapp.azure.com/return_429";
+
+/**
+*  This function is the main entry of the DfciCheck429 application.
+*
+* @param[in]   ImageHandle
+* @param[in]   SystemTable
+*
+**/
+EFI_STATUS
+EFIAPI
+DfciCheck429Entry (
+  IN EFI_HANDLE        ImageHandle,
+  IN EFI_SYSTEM_TABLE  *SystemTable
+  )
+{
+  EFI_STATUS  Status;
+
+  ZeroMem (&mDfciNetworkRequest, sizeof (mDfciNetworkRequest));
+
+  mDfciNetworkRequest.HttpRequest.Url = AllocateCopyPool (sizeof (mCheck429Url), &mCheck429Url);
+  mDfciNetworkRequest.MainLogic       = Check429Logic;
+
+  // Try every NIC in the system until one fills the first part of the request.
+  Status = TryEachNICThenProcessRequest (&mDfciNetworkRequest);
+
+  DEBUG ((DEBUG_INFO, "Url   %a\n", mDfciNetworkRequest.HttpRequest.Url));
+  DEBUG ((DEBUG_INFO, "HttpStatus = %d, Status-%r\n", mDfciNetworkRequest.HttpStatus.HttpStatus, Status));
+  DEBUG ((DEBUG_INFO, "HttpStatus = %a\n", GetHttpErrorMsg (mDfciNetworkRequest.HttpStatus.HttpStatus)));
+

```

Comment:
Should we check that TryEachNICThenProcessRequest returned success before we access the data?

Reply:
Fixed